docs/general/naming.md

```diff
@@ -309,4 +315,106 @@ freely. For example, `system.paging.faults` and `system.network.packets`.
 
+### Client and Server Metrics
+
+Metrics that measure some aspect of a physical or logical network calls SHOULD include
+an indication of the side metric being recorded.
+
+Such metrics SHOULD follow the `{area}.{client|server}.{metric_name}`
+pattern if communication side is ambiguous for given `{area}` and `{metric_name}`.
+Otherwise, when a specific communication side is implied by given `{area}` or
+`{metric_name}`, the `{area}.{metric_name}` pattern SHOULD be used.
+
+Examples:
+- `http.client.request.duration`
+- `gen_ai.server.request.duration`
+- `messaging.client.sent.messages`
+- `messaging.process.duration` - the term `process` clearly indicates that
+  the metric is reported by the consumer.
+- `kestrel.connection.duration` - here, `kestrel` is the name of the web server,
+  so no additional indication is necessary.
+
+## System-specific naming
+
+**Status**: [Development][DocumentStatus]
+
+### System (project/product/provider) name attribute
+
+Semantic conventions for a certain area are usually applicable to multiple systems
+(projects, providers, products).
+
+For example, database semantic conventions can be used to describe telemetry for a
+broad range of database systems.
+
+Such conventions SHOULD define an attribute to represent system name following
+`{area}.system|provider|protocol.name` pattern.
+
+For example, database conventions include the `db.system.name` attribute.
+<!-- not yet, we should not merge it before https://github.com/open-telemetry/semantic-conventions/pull/1613 -->
+
+### Choosing a System Name
+
+When adding new system to the semantic conventions, follow these principles in descending order of priority:
+
+1. The system name SHOULD adhere to the general attribute naming guidelines outlined in this document,
+   as it will be used as a namespace in [system-specific attribute names](#system-specific-attributes).
+
+2. The system name SHOULD unambiguously identify this specific product or project.
+
+   For example, use `gcp.pubsub` or `oracle.db`. Avoid generic names like `pubsub`,
+   which could refer to multiple messaging products, or `oracle` which lacks specificity.
+
+3. The system name SHOULD match the corresponding project or product name in the following cases:
+
+   - Independent projects that do not belong to a specific company, such as Apache Foundation projects like
+     `kafka` or `cassandra`.
+   - Products with names similar to the owning company, such as `mongodb` or `elasticsearch`
+   - Widely recognized products that are popular outside their company's ecosystem.
+     These products often have trademarks without the company name and have
+     their own top-level domain (e.g. `spring` or `mysql`).
+
+3. In other cases, the system name SHOULD be prefixed with the company (organization,
+   division, or group) name. For cloud services, the name SHOULD use the
+   corresponding cloud provider name. For example, use `ibm.db2` or `aws.dynamodb`.
```

Comment:
`ibm.db2` might not be the best example since some (me) don't think of db2 primarily as a cloud service (i'm probably wrong at this point...), and it's not clear to me whether ibm cloud will have its own namespace (`ibm.cloud`?)

```suggestion
   corresponding cloud provider name. For example, use `aws.dynamodb` or `azure.cosmosdb`.
```